### DISCUSSION TITLE
[help] Examples of content adressable storage with AWS?

### DISCUSSION THREAD

**Discussion by Aariq** [ropensci/targets#1387] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I'm wondering if anyone (E.g. @noamross) has any examples of using `tar_repository_cas()` to use CAS with S3 buckets?  I am hoping to test it as a workaround to get S3 storage working for `geotargets` where the download function would just return a path that `terra::rast()` could use directly rather than actually downloading the object (https://github.com/ropensci/targets/discussions/1367).  However, I don't know how to write the other required functions and I'm hoping to find some working examples to build off of.

**Comment by wlandau** [comment]
CAS in `targets` is new enough that I am not aware of any S3 examples, but you could borrow from the utility functions in https://github.com/ropensci/targets/blob/main/R/utils_aws.R to implement one.
